### PR TITLE
Stream a packet to all peers using only one single file

### PR DESCRIPTION
**Pull request by sebtomba**
## Overview
Stream a packet to all peers using only one single file

### JIRA ticket:
https://rchain.atlassian.net/browse/RCHAIN-2793

### Please make sure that this PR:
- [x] is at most 200 lines of code (excluding tests),
- [x] meets [RChain development coding standards](https://rchain.atlassian.net/wiki/spaces/DOC/pages/28082177/Coding+Standards),
- [x] includes tests for all added features,
- [x] has a reviewer assigned,
- [x] has [all commits signed](https://rchain.atlassian.net/wiki/spaces/DOC/pages/498630673/How+to+sign+commits+to+rchain+rchain).

### CODE REVIEW

**Comment by rabbitonweb** [comment]
The more I read PR's that operate on files, the more I think we need `FileSystem` algebra (in tagless final encoding). PR looks good so far.

File: comm/src/main/scala/coop/rchain/comm/transport/StreamObservable.scala [hunk 1]
```diff
@@ -11,38 +11,41 @@ import monix.reactive.Observable
...
-          log.error(s"Could not serialize packet ${blob.packet}. Error: $er") *> true.pure[Task]
+          log.error(s"Could not serialize packet ${blob.packet}. Error: $er") *> None.pure[Task]
       }
 
-    def retry(failed: List[PeerNode]): Task[Unit] =
+    def push(file: Path): Task[Boolean] =
+      Task.delay(subject.pushNext(StreamToPeers(peers, file, blob.sender)))
+
+    def propose(file: Path): Task[Unit] =
+      push(file) >>= (_.fold(Task.unit, retry(file)))
```
Comment (rabbitonweb) [inline comm/src/main/scala/coop/rchain/comm/transport/StreamObservable.scala]: I know this is unrelated to this PR, but isn't that a potential infinite loop? we need to stop at some point?

File: comm/src/main/scala/coop/rchain/comm/transport/TcpTransportLayer.scala [hunk 2]
```diff
@@ -208,34 +208,29 @@ class TcpTransportLayer(
   def broadcast(peers: Seq[PeerNode], msg: Protocol): Task[Seq[CommErr[Unit]]] =
     innerBroadcast(peers, msg)
 
-  def handleToStream(toStream: ToStream): Task[Unit] = {
+  private def streamToPeer(peer: PeerNode, path: Path, sender: PeerNode): Task[Unit] = {
```
Comment (rabbitonweb) [inline comm/src/main/scala/coop/rchain/comm/transport/TcpTransportLayer.scala]: name is misleading, `handle` convention was always good, you knew you are now "handling" a message, you are "on the server side"

File: comm/src/main/scala/coop/rchain/comm/transport/StreamObservable.scala (hunk 1 above)
Comment (sebtomba) [reply comm/src/main/scala/coop/rchain/comm/transport/StreamObservable.scala]: Yes, I had the same concern. On the other hand we are pushing an item to a queue. Eventually the queue will accept the item. Until then Casper must be blocked and not produce new items. We have limited reties (ATM 3x) on the other end of the message queue. So `propose`/`retry` should work as expected and not run forever.

File: comm/src/main/scala/coop/rchain/comm/transport/TcpTransportLayer.scala (hunk 2 above)
Comment (sebtomba) [reply comm/src/main/scala/coop/rchain/comm/transport/TcpTransportLayer.scala]: I don't agree. We are not handling a message (`ToStream') here anymore. It's a function with parameters. We are on the client side and want to stream a file to some peer node (server side). Maybe I can change the parameter name `path` to `file`?

**Pull request merged**